features/FEATURE_BLE/ble/gap/ConnectionParameters.h

```diff
@@ -0,0 +1,465 @@
+        );
+    }
+
+#if !defined(DOXYGEN_ONLY)
+
+    uint8_t getPhySet() const
+    {
+        phy_set_t set(
+            _enabledPhy[LE_1M_INDEX],
+            _enabledPhy[LE_2M_INDEX],
+            _enabledPhy[LE_CODED_INDEX]
+        );
+        return set.value();
+    }
+
+    /* these return pointers to arrays of settings valid only across the number of active PHYs */
+
+    const uint16_t *getScanIntervalArray() const
+    {
+        return &_scanInterval[getFirstEnabledIndex()];
+    }
+
+    const uint16_t *getScanWindowArray() const
+    {
+        return &_scanWindow[getFirstEnabledIndex()];
+    }
+
+    const uint16_t *getMinConnectionIntervalArray() const
+    {
+        return &_minConnectionInterval[getFirstEnabledIndex()];
+    }
+
+    const uint16_t *getMaxConnectionIntervalArray() const
+    {
+        return &_maxConnectionInterval[getFirstEnabledIndex()];
+    }
+
+    const uint16_t *getSlaveLatencyArray() const
+    {
+        return &_slaveLatency[getFirstEnabledIndex()];
+    }
+
+    const uint16_t *getConnectionSupervisionTimeoutArray() const
+    {
+        return &_connectionSupervisionTimeout[getFirstEnabledIndex()];
+    }
+
+    const uint16_t *getMinEventLengthArray() const
+    {
+        return &_minEventLength[getFirstEnabledIndex()];
+    }
+
+    const uint16_t *getMaxEventLengthArray() const
+    {
+        return &_maxEventLength[getFirstEnabledIndex()];
+    }
+
+#endif
+
+private:
+    uint8_t getFirstEnabledIndex() const
+    {
+        if (_enabledPhy[LE_1M_INDEX]) {
+            return 0;
+        } else if (_enabledPhy[LE_2M_INDEX]) {
+            return 1;
+        } else if (_enabledPhy[LE_CODED_INDEX]) {
+            return 2;
+        }
+        /* this should never happen, it means you were trying to start a connection with a blank set
+         * of paramters - you need to enabled at least one phy */
+        MBED_ASSERT("Trying to use connection parameters without any PHY defined.");
+        return 0;
+    }
+
+    /** Handle toggling PHYs on and off and return the correct index to use to set the configuration elements.
+     *
+     * @param phy Which Phy is being toggle.
+     * @param enable On or Off.
+     * @return The index to the array of settings.
+     */
+    uint8_t handlePhyToggle(phy_t phy, bool enable)
+    {
+        uint8_t index = phyToIndex(phy);
+
+        bool was_swapped = false;
+        bool is_swapped = false;
+
+        if (isSwapped()) {
+            was_swapped = true;
+        }
+
+        _enabledPhy[index] = enable;
+
+        if (isSwapped()) {
+            is_swapped = true;
+        }
+
+        if (was_swapped != is_swapped) {
+            swapCodedAnd2M();
+        }
+
+        if (is_swapped && phy == phy_t::LE_CODED) {
+            index -= 1;
+        }
+
+        return index;
+    }
+
+    uint8_t phyToIndex(phy_t phy) const
+    {
+        uint8_t index;
+        switch(phy.value()) {
+            case phy_t::LE_1M:
```

Comment:
will be handled by default case

Reply:
Yes I understand that, my question was more to do with , why bother to add the NONE type in the first place if it isn't used ?

Reply:
NONE is not a valid phy therefore there is no index for it in this array which only stores possible phys.

(BTW the reason for the the elaborate swapping and indexing is because the underlying layer requires individual arrays of parameter so they need to be contiguous in memory)